tests/slow/src/test/scala/tests/mill/MillServerSuite.scala

```diff
@@ -49,25 +51,36 @@ class MillServerSuite
     }
   }
 
-  test("generate") {
-    def millBspConfig = workspace.resolve(".bsp/mill-bsp.json")
-    cleanWorkspace()
-    writeLayout(MillBuildLayout("", V.scala213, supportedBspVersion))
-    for {
-      _ <- server.initialize()
-      _ <- server.initialized()
-      _ = assertNoDiff(
-        client.workspaceMessageRequests,
-        // Project has no .bloop directory so user is asked to "import via bloop"
-        // since bloop is still the default
-        importBuildMessage,
-      )
-      _ = client.messageRequests.clear() // restart
-      _ = assert(!millBspConfig.exists)
-      // At this point, we want to use mill-bsp server, so create the mill-bsp.json file.
-      _ <- server.executeCommand(ServerCommands.GenerateBspConfig)
-    } yield {
-      assert(millBspConfig.exists)
+  val versionsToTest: List[String] =
+    List("0.10.0", "0.10.8", supportedBspVersion)
+
+  versionsToTest.foreach(testGenerationAndConnection)
+
+  private def testGenerationAndConnection(version: String) = {
+    test(s"generate-and-connect-$version") {
+      def millBspConfig = workspace.resolve(".bsp/mill-bsp.json")
+      cleanWorkspace()
+      writeLayout(MillBuildLayout("", V.scala213, version))
+      for {
+        _ <- server.initialize()
+        _ <- server.initialized()
+        _ = assertNoDiff(
+          client.workspaceMessageRequests,
+          // Project has no .bloop directory so user is asked to "import via bloop"
+          // since bloop is still the default
+          importBuildMessage,
+        )
+        _ = client.messageRequests.clear() // restart
+        _ = assert(!millBspConfig.exists)
+        // At this point, we want to use mill-bsp server, so create the mill-bsp.json file.
+        _ <- server.executeCommand(ServerCommands.GenerateBspConfig)
+        // We need to wait a bit just to ensure the connection is made
+        _ <- server.waitFor(TimeUnit.SECONDS.toMillis(3))
```

Comment:
Might be better to just wait for the connect, otherwise it could be flaky.
```suggestion
        _ <- server.server.buildServerPromise.future
```